django/db/models/functions/math.py

```diff
@@ -0,0 +1,184 @@
+import math
+import sys
+
+from django.db.models import DecimalField, FloatField, Func, Transform
+from django.db.models.functions import Cast
+
+
+class DecimalInputMixin:
+
+    def as_postgresql(self, compiler, connection):
+        # Cast FloatField to DecimalField as PostgreSQL doesn't support the
+        # following function signatures by default:
+        # - LOG(double precision, double precision)
+        # - MOD(double precision, double precision)
+        output_field = DecimalField(decimal_places=sys.float_info.dig, max_digits=1000)
+        clone = self.copy()
+        clone.set_source_expressions([
+            Cast(expression, output_field) if isinstance(expression.output_field, FloatField)
+            else expression for expression in self.get_source_expressions()
+        ])
+        return clone.as_sql(compiler, connection)
+
+
+class OutputFieldMixin:
+
+    def _resolve_output_field(self):
+        sources = self.get_source_expressions()
+        if any(isinstance(s.output_field, DecimalField) for s in sources):
+            return DecimalField()
+        else:
+            return FloatField()
+
+
+class Abs(Transform):
+    function = 'ABS'
+    lookup_name = 'abs'
+
+
+class ACos(OutputFieldMixin, Transform):
+    function = 'ACOS'
+    lookup_name = 'acos'
+
+
+class ASin(OutputFieldMixin, Transform):
+    function = 'ASIN'
+    lookup_name = 'asin'
+
+
+class ATan(OutputFieldMixin, Transform):
+    function = 'ATAN'
+    lookup_name = 'atan'
+
+
+class ATan2(OutputFieldMixin, Func):
+    function = 'ATAN2'
+    arity = 2
+
+    def as_sqlite(self, compiler, connection):
+        if not getattr(connection.ops, 'spatialite', False):
+            return self.as_sql(compiler, connection)
+        if connection.ops.spatial_version < (4, 3, 0):
+            return self.as_sql(compiler, connection)
+        # As documented this function ought to be ATan2(y, x) returning the
+        # inverse tangent of y / x, but it turns out that, unlike all other
+        # backends, the function is actually ATan2(x, y). In addition, it
+        # doesn't handle integer inputs properly, so these need to be casted to
+        # floats.
+        # Although the spatialite backend inherits from the sqlite3 backend,
+        # the spatialite function takes precedence over the one created using
+        # create_function().
+        clone = self.copy()
+        clone.set_source_expressions([
+            Cast(expression, FloatField()) if not isinstance(expression.output_field, FloatField)
```

Comment:
Could you also change this line for me?
```diff
-            Cast(expression, FloatField()) if not isinstance(expression.output_field, FloatField)
+            Cast(expression, FloatField()) if isinstance(expression.output_field, IntegerField)
```
Don't forget to import `IntegerField`.